alert-database/src/main/java/com/synopsys/integration/alert/database/configuration/repository/ConfigGroupRepository.java

```diff
@@ -26,11 +26,41 @@
 import java.util.UUID;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Component;
 
 import com.synopsys.integration.alert.database.configuration.ConfigGroupEntity;
 
 @Component
 public interface ConfigGroupRepository extends JpaRepository<ConfigGroupEntity, Long> {
     List<ConfigGroupEntity> findByJobId(UUID jobId);
+
+    @Query(value = "SELECT job FROM ConfigGroupEntity job WHERE job.jobId IN :jobIds")
+    List<ConfigGroupEntity> findByJobIds(@Param("jobIds") List<UUID> jobIds);
+
+    @Query(value = "SELECT cast(job_with_field_count.job_id as varchar) as job_id FROM (SELECT job.job_id, COUNT(fieldValues.id) AS matching_field_count"
+                       + "    FROM alert.config_groups job"
+                       + "    LEFT JOIN alert.descriptor_configs descConf ON job.config_id = descConf.id"
+                       + "    LEFT JOIN alert.field_values fieldValues ON descConf.id = fieldValues.config_id"
+                       + "    WHERE (fieldValues.field_id = GET_FIELD_ID('channel.common.frequency') AND fieldValues.field_value = :frequency)"
+                       + "    OR (fieldValues.field_id = GET_FIELD_ID('provider.common.config.name') AND fieldValues.field_value = :providerConfigName)"
```

Comment:
Unless I'm just not seeing it, this will not account for project name patterns.

Reply:
Looks like @gkillough is right. I don't see anything in the query that would account for project names.  We would need a `LIKE` or `REGEXP` expression.

Reply:
Nvm, I thought this was something different.